src/includes.yaml

```diff
@@ -127,6 +198,154 @@ templates:
         - name: today
           comment: Scheduled for _today_.
 
+  - template: &PropType.External
+    - type: Map
+      example_values:
+        - "{ id: 'my_id', data: 'A blob of information.' }"
+      access: Oauth Required
+
+  - template: &PropType.TaskName
+    - type: String
+      example_values:
+        - "'Buy catnip'"
+
+  - template: &PropType.TaskNotes
+    - type: String
+      example_values:
+        - "'Mittens really likes the stuff from Humboldt.'"
+
+  - template: &PropType.ProjectArray
+    - type: Array
+      example_values:
+        - "[ { id: 1331, name: 'Stuff to Buy' }, ... ]"
+      access: Create-only
+
+  - template: &PropType.Memberships
+    - type: Array
+      example_values:
+        - "[ { project: { id: 1331, name: 'Bugs' }, section: { id: 1123, name: 'P1:' } }, ... ]"
+      access: Create-only
+
+  ## Project/Tag
+  - template: &PropType.ProjectName
```

Comment:
So, the name of these templatized properties begin with `PropType` which signifies "property type". The set of things that was in here initially was at a higher level of abstraction than individual properties, they were "types" of properties. So, `Bool` for a boolean, `User` for any user reference, etc.

This makes it a little less clear what all the new templates you've added are for. It seems totally reasonable to have each property itself be in a template, because we may be referencing that property from multiple places (if that's what you're going for?). But we should be careful to separate them from property types.

So for example, "project name" is a property, not a property type. The type is really just a string. So we might want instead something like:

    - template: &Project.Name
      ...

The property type templates are useful because each type has its own example value and some types are complicated so the example values are annoying. We probably want to have the property _instance_ templates inherit from the property _type_ templates, for the most part.

Also, if we're making templates per property per resource, and each one only shows up in a single resource, perhaps the templates should just exist in the resource file itself?